openhtf/util/logs.py

```diff
@@ -88,6 +89,20 @@ def MyRandomMethod():
 LOGGER_PREFIX = 'openhtf'
 RECORD_LOGGER = '.'.join((LOGGER_PREFIX, 'test_record'))
 
+_LOGONCE_SEEN = set()
+
+LogRecord = collections.namedtuple(
+    'LogRecord', 'level logger_name source lineno timestamp_millis message')
+
+
+def LogOnce(log_func, msg, *args, **kwargs):
+  """"Logs a message only once."""
+  if msg not in _LOGONCE_SEEN:
+    log_func(msg, *args, **kwargs)
+    # Only check the message since it's likely from the source code so lifespan
```

Comment:
i'm not a huge fan of keying on the message, would rather key on the stack frame that it was called from or something, but i have a feeling that's going to be annoying to implement, so let's just punt for now, this looks good enough, thanks!